update: AI/ML Inferentia lab also uses Trainium instances

bkgardiner opened the pull request:
#### What this PR does / why we need it:
PR adds support for Trn1 instances and updates the module to the latest version of Neuron SDK.

#### Which issue(s) this PR fixes:
Clean up script now cleans up regardless of where you are in the module. 
Updated to Karpenter V1

Fixes #

#### Quality checks

- [X] My content adheres to the style guidelines
- [X] I ran `make test module="<module>"` it was successful (see https://github.com/aws-samples/eks-workshop-v2/blob/main/docs/automated_tests.md)

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

svennam92 commented on website/docs/aiml/inferentia/karpenter.md:
````diff
@@ -0,0 +1,44 @@
+karpenter   2/2     2            2           11m
+```
+
+The only setup that we will need to do is to update our EKS IAM mappings to allow Karpenter nodes to join the cluster:
````
we no longer need the iamidentitymapping because we're using pod identity

bkgardiner commented:
I've removed the identity mapping and tested Karpenter. I've also moved the container image environment variables to the Terraform file so users can't accidentally use the wrong images.
